src/stan/variational/advi.hpp

```diff
@@ -806,11 +806,20 @@ class advi {
     bool khat_failed = false;
     for(int k = 0; k < num_chains; k++){
       std::stringstream ss;
-      Eigen::VectorXd iw_vec(n_posterior_samples_);
-      lr(variational_obj_vec[k], iw_vec);
-      //ss << iw_vec;
+      Eigen::VectorXd lw_vec(n_posterior_samples_);
+      lr(variational_obj_vec[k], lw_vec);
       double sigma;
-      stan::analyze::gpdfit(iw_vec, khat, sigma);
+      int n_tail;
+      if(n_posterior_samples_ < 255) {
+        n_tail = int(lw_vec.size() * 0.2);// if more than 255 samples 3 * sqrt(lw_vec.size())
+      }else{
+        n_tail = 3 * sqrt(lw_vec.size());// if more than 255 samples
+      }
+      Eigen::VectorXd w_tail(n_tail);
+      for (int n = 0; n < n_tail; ++n) {
+        w_tail(n) = exp(lw_vec(lw_vec.size() - n_tail + n)) - exp(lw_vec(lw_vec.size() - n_tail));
+      }
+      stan::analyze::gpdfit(w_tail, khat, sigma);
```

Comment:
added [this](https://github.com/stan-dev/loo/blob/51fb2d0511a765407ff1bd47fab7e79b5c62cbfb/R/psis.R#L209) logic from loo package, but getting minus khat values as follows:
```
Chain 0khat: 0.305205
Chain 1khat: -0.8578
Chain 2khat: 0.194817
Chain 3khat: -0.531074
```

Reply:
If you need the last ` n_tail` values in a vector,  you can use ` lw_vec.tail(n_tail)`